libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorPipeChannelTest.java

```diff
@@ -18,14 +18,15 @@
 import java.io.IOException;
 import java.nio.channels.Pipe;
 
-class FIXInitiatorPipeChannelTest extends FIXInitiatorTest<PipeChannel> {
+class FIXInitiatorPipeChannelTest extends FIXInitiatorTest {
     @Override
     protected Channels createChannels() throws IOException {
         Pipe up = Pipe.open();
         Pipe down = Pipe.open();
 
         Channels channels = new Channels();
-        channels.initiator = new PipeChannel(up.sink(), down.source());
+        channels.initiatorRead = down.source();
+        channels.initiatorWrite = up.sink();
```

Comment:
Note that we are not using the `PipeChannel` for the `FIXConnection`, and we could have rewritten the acceptor class to also not need it (if we segregate the upstream and downstream channels).

Reply:
If we can get rid of `PipeChannel` as an abstraction, then I think it would be worthwhile to do the change also on the acceptor side.